src/bootstrap/compile.rs

```diff
@@ -474,17 +475,19 @@ pub fn tool(build: &Build, stage: u32, target: &str, tool: &str) {
         cargo.env("LIBZ_SYS_STATIC", "1");
     }
 
-    cargo.env("CFG_RELEASE_CHANNEL", &build.config.channel);
+    if stage == 2 {
```

Comment:
Could we leave this as is? I don't believe there's any reason to omit this information in other stages?